codes/quantum/oscillators/fock_state/knill-laflamme-milburn-loqc.yml

```diff
@@ -0,0 +1,71 @@
+    - 'The dual-rail logic encoding is isomorphic to encoding in the horizontal and vertically polarized basis, up to a transformation using a polarising beam-splitter and polarising rotator.'
+    - 'The dual-rail code is concatenated with an outer stabilizer code, with the following two codewords: Logical \(|\bar{0}\rangle\) is encoded as \(\left|\Phi^{+}\right\rangle = \frac{1}{2}(|00\rangle_q + |11\rangle_q)\), and logical \(|\bar{1}\rangle\) is encoded as \(\left|\Psi^{+}\right\rangle  = \frac{1}{2}(|01\rangle_q + |10\rangle_q)\).'
+
+  transversal_gates: 'The KLM protocol relies on single-qubit unitaries or controlled-Z gates, which are transversal. Hence the encoded operations can be performed transversally.'
+
+  general_gates:
+    - 'Single-qubit unitaries can be performed using purely linear optical elements. This is because any single qubit unitary U can be decomposed into rotations about the Z and Y axis in the Bloch sphere i.e. \( U=e^{i \alpha} R_{z}(\beta) R_{y}(\gamma) R_{z}(\delta)\). Arbitary Z and Y rotations which can be achieved using phase-shifters and beam splitters.'
+    - 'Universal gate set achieved by adding the controlled-Z phase (CZ) gate to the set of single-qubit unitaries.'
+    - 'A probabilistic controlled-Z gate is implemented by making use of the non-linear sign shift \((NS_{-1})\) gate, and then measuring the circuit adaptively to approximate linearity. The non-linear sign shift gate is defined by the following action: \( \alpha|0\rangle+\beta|1\rangle+\gamma|2\rangle \rightarrow \alpha|0\rangle+\beta|1\rangle-\gamma|2\rangle \)'
+    - 'The non-linear sign-shift gate \((NS_{-1})\) can be performed with a success probability of 1/4 \cite{arXiv:quant-ph/0512104}'
+    - 'Two non-linear sign-shift gates can perform the C-Sign gate with success probability 1/16. However it is not known whether these '
+    - 'Knill has shown that the CSign gate has an upper bound of 3/4 when using two ancilla modes, each with a maximum of 1 photon per mode.\cite{doi:10.1103/PhysRevA.68.064303}'
+    - 'The probability of success of the CZ gate (referred to in \cite{doi:10.1038/35051009} as the C-Sign gate) is improved using teleportation, by reducing the problem of construction of 2-qubit gates to a state preparation problem. We know when the state preparation fails, and can perform a correction.'
+    - 'The success probability of the CZ gate is further improved by using Z-measurement error-correction during teleportation.'
+
+  decoders:
+    - 'In case of Z-measurement errors i.e. \(Z \otimes \mathbb{1}\) and \(\mathbb{1} \otimes Z \), we perform syndrome measurement by measuring \(X \otimes X\), and apply Z-operation on the first and/or second qubit conditioned on the outcomes.'
```

Comment:
This is for the specific bell state encoding?

Reply:
That's a good catch, I'll preface this with a more general remark, and reframe the bell encoding as an example for a specific case